api/services/dataset_service.py

```diff
@@ -319,6 +320,22 @@ def update_dataset(dataset_id, data, user):
                     data["embedding_model_provider"] != dataset.embedding_model_provider
                     or data["embedding_model"] != dataset.embedding_model
                 ):
+                    indexing_documents = (
+                        db.session.query(Document)
+                        .filter(
+                            Document.dataset_id == dataset_id,
+                            Document.indexing_status == "indexing",
+                        )
+                        .all()
+                    )
+                    if len(indexing_documents) > 0:
```

Comment:
Loading all the docs into memory is risky for huge memory usage. Counting the docs on db side instead.